pkg/acsengine/Get-AzureConstants.py

```diff
@@ -0,0 +1,241 @@
+
+def getAllSizes():
+    command = "az account list-locations"
+    proc = subprocess.Popen(command, stdout=subprocess.PIPE, shell=True)
+    (locations, err) = proc.communicate()
+    if err:
+        raise Exception(err.message)
+    
+    sizeMap = {}
+    locations = json.loads(locations)
+    for location in locations:
+        command = "az vm list-sizes -l %s" % (location['name'])
+        proc = subprocess.Popen(command , shell=True, stdout=subprocess.PIPE)
+        (sizes, err) = proc.communicate()
+        if err:
+            raise Exception(err.message)
+        
+        sizes = json.loads(sizes)
+        for size in sizes:
+            if not size['name'] in sizeMap and not size['name'].split('_')[0] == 'Basic':
+                sizeMap[size['name']] = size
+
+    return sizeMap        
+
+min_cores = 2
+dcos_masters_ephemeral_disk_min = 102400
+
+def getDcosMasterMap(sizeMap):
+    masterMap = {}
+
+    for key in sizeMap.keys():
+        size = sizeMap[key]
+        if size['numberOfCores'] >= min_cores and \
+           size['resourceDiskSizeInMb'] >= dcos_masters_ephemeral_disk_min:
+            masterMap[size['name']] = size
+
+    return masterMap    
+
+def getMasterAgentMap(sizeMap):
+    agentMap = {}
+    
+    for key in sizeMap.keys():
+        size = sizeMap[key]
+        if size['numberOfCores'] >= min_cores:
+            agentMap[size['name']] = size
+    
+    return agentMap
+
+def getLocations():
+    proc = subprocess.Popen(['az account list-locations'], stdout=subprocess.PIPE, shell=True)
+    (locations, err) = proc.communicate()
+    if err:
+        raise Exception(err.message)
+
+    locations = json.loads(locations)
+    locationList = []
+    for location in locations:
+        locationList.append(location['name'])
+
+    #hard code Azure China Cloud location
+    locationList.append('chinanorth')
+    locationList.append('chinaeast')
+
+    locationList = sorted(locationList)
+    return locationList    
+
+def getStorageAccountType(sizeName):
+    capability = sizeName.split('_')[1]
+    if 'S' in capability or 's' in capability:
+        return "Premium_LRS"
+    else:
+        return "Standard_LRS"
+
+def getFileContents(dcosMasterMap, masterAgentMap, kubernetesAgentMap, sizeMap, locations):
+    text = 'package acsengine'
+    text += '\n\nimport "fmt"'
+
+    text += '\
+\n\n\
+// AUTOGENERATED FILE - last generated ' + time + '\n\n\
+const (\n\
+        // AzurePublicProdFQDNFormat specifies the format for a prod dns name \n\
+        AzurePublicProdFQDNFormat = "%s.%s.cloudapp.azure.com"\n\
+        //AzureChinaProdFQDNFormat specify the endpoint of Azure China Cloud\n\
+        AzureChinaProdFQDNFormat = "%s.%s.cloudapp.chinacloudapi.cn"\n\
+)\n\n'
+
+    text += '// AzureLocations provides all azure regions in prod.\n\
+// Related powershell to refresh this list:\n\
+//   Get-AzureRmLocation | Select-Object -Property Location\n\
+var AzureLocations = []string{\n'

```

Comment:
Generally, I still wish that this only generated a single file with only constants in it. I don't understand why we're spitting out static functions in a generated file (FormatAzureProdFQDNs, why are the FQDN formats in this file, etc)